### ISSUE TITLE
control lemma creation of new glob

### ISSUE THREAD

**Issue by arademaker** [own-pt/sensetion.el#80]
allow only lemas in the WN. A glob can only be created if it exists in WN.

**Comment by odanoburu** [comment]
I disagree with this one; if we decide that something a MWE but isn't in WN
we won't be able to glob it?

if we do allow it, we can even find new sense candidates by searching for
globs whose senses are not in WN.

>

**Comment by arademaker** [comment]
> On 28 Mar 2019, at 15:34, bruno cuconato <notifications@github.com> wrote:
> 
> I disagree with this one; if we decide that something a MWE but isn't in WN
> we won't be able to glob it?

What would be the criteria for globing? That is my point. MWE are hard. We do have other ways to decide about them during parsing and than we first decide about them, add in the WN before allow decision during annotation.

> if we do allow it, we can even find new sense candidates by searching for
> globs whose senses are not in WN.
> 

It will end up with a messing data. We may commit a mistake in the lemma, and we can make inconsistent decisions about what should be a glob.

**Comment by odanoburu** [comment]
I see your point, even though I don't agree 100% on it. It is implemented, but how should we handle the lemma editing command then? that allows any kind of edit to happen...

**Comment by arademaker** [comment]
Hum... I see your point too. We should make this behavior something that we can configure. Let us call this restricted mode vs relaxed mode of operation. Given that, in the restricted mode, yes, any lemma must be in the lexicon. You can also enforce the format LEMMA%X where X can be 1, 2, 3 or 4. In the relaxed mode, we are free to add lemmas that are not in the lexicon.  

Works for you? 

I myself will prefer to work on the restricted mode for a while...

**Comment by arademaker** [comment]
Is it implemented? BTW, it is important to enforce the lemma%POS format , I always forget the POS. BTW, in the future, we should try to check (fuzzy search? distance edition?) if the given lemma is the expected one.

**Comment by odanoburu** [comment]
no, the modes are not implemented, the interface works as if in restricted mode. what else would such modes entail? because for only this small behaviour change I think it is uneeded complexity..

**Comment by arademaker** [comment]
Are you sure? `restricted mode`? No, i am using the sensetion.el from the master of this repo, when I glob something, it is not imposing any constraint on the lemma or POS tag.

**Comment by odanoburu** [comment]
it will only complete lemmas that are already in the index, but tbh I'm not sure if it will complain if you manually enter something else. and it doesn't specify a POS tag at all.

I'll fix it, but I'd rather focus on the new version first. unless you disagree, of course.

**Comment by arademaker** [comment]
no, go ahead. I will make explicit when something is urgent.

**Comment by odanoburu** [comment]
you were right, it was not operating in strict mode, mainly because the lemma field was a string, and this string could represent several actual lemmas. the old command would substitute whatever was typed as the new lemma, and this did not lend itself to completion. 

the changes being implemented in the `es` branch involve making the lemma field a list of strings; now the lemma editing command adds a new lemma (leaving the old ones intact), including its POS. this was done in https://github.com/own-pt/sensetion.el/commit/c0d84aa203a0e21e3692ba1d536155aec6786c62

**Comment by arademaker** [comment]
OK, now I remember, we have to think better about how to impose some verification on the lemmas. But what is the plan about the branch `es`? Note that changes must be always consider carefully, for instance, if we move from a string to a list of strings, we need to support all operations on lists, right? At least, adding items an removing items.

**Comment by odanoburu** [comment]
I see a few ways out of this:

- we had thought about not allowing any kind of editing different from sense annotation, and that any other problems would be corrected at the parsing stage
- we can keep lemmas as strings, I can have completion even in this way, but I see no way of preventing the user from inputting an invalid lemma, and they would have to encode the pos themselves (`be` -> `be%2`)
- we can add all the list operations

**Comment by odanoburu** [comment]
ping @arademaker

**Comment by odanoburu** [comment]
sorry, new features are in the `es` branch.

yes, this is a tentative fix; please tell me what you think.

they are different in that the lemma of an existing world we are editing pre-existing information which may include several lemmas+pos. in the case of globbing, we are creating new data by asking for one lemma and pos.
